src/test/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleTest.java

```diff
@@ -210,17 +188,91 @@ public void shouldCreateChartsAndAccessThemViaInteractiveQueries() throws Except
+    appServerPort = randomFreeLocalPort();
+    streams =
+        KafkaMusicExample.createChartsStreams(CLUSTER.bootstrapServers(),
+                                              CLUSTER.schemaRegistryUrl(),
+                                              appServerPort,
+                                              TestUtils.tempDirectory().getPath(),
+                                              host);
+    restProxy = KafkaMusicExample.startRestProxy(streams, new HostInfo(host, appServerPort));
+  }
+
+  @After
+  public void shutdown() throws Exception {
+    restProxy.stop();
+    streams.close();
+  }
 
+  @Test
+  public void shouldCreateChartsAndAccessThemViaInteractiveQueries() throws Exception {
+    final String host = "localhost";
+    createStreams(host);
     streams.start();
 
+    // wait until the StreamsMetadata is available as this indicates that
+    // KafkaStreams initialization has occurred
+    TestUtils.waitForCondition(() -> !StreamsMetadata.NOT_AVAILABLE.equals(streams.allMetadataForStore(KafkaMusicExample.TOP_FIVE_SONGS_STORE)),
+                               MAX_WAIT_MS,
+                               "StreamsMetadata should be available");
+    final String baseUrl = "http://" + host + ":" + appServerPort + "/kafka-music";
+    final Client client = ClientBuilder.newClient();
+
+    // Wait until the all-songs state store has some data in it
+    TestUtils.waitForCondition(() -> {
+      final ReadOnlyKeyValueStore<Long, Song>
+          songsStore;
+      try {
+        songsStore =
+            streams.store(KafkaMusicExample.ALL_SONGS, QueryableStoreTypes.<Long, Song>keyValueStore());
+        return songsStore.all().hasNext();
+      } catch (Exception e) {
+        return false;
+      }
+    }, MAX_WAIT_MS, KafkaMusicExample.ALL_SONGS + " should be non-empty");
+
+    final IntFunction<SongPlayCountBean> intFunction = index -> {
+      final Song song = songs.get(index);
+      return songCountPlayBean(song, 6L - (index % 6));
+    };
+
+    // Verify that the charts are as expected
+    verifyChart(baseUrl + "/charts/genre/punk",
+                client,
+                IntStream.range(0, 5).mapToObj(intFunction).collect(Collectors.toList()));
+
+    verifyChart(baseUrl + "/charts/genre/hip hop",
+                client,
+                IntStream.range(6, 11).mapToObj(intFunction).collect(Collectors.toList()));
+
+    verifyChart(baseUrl + "/charts/top-five",
+                client,
+                Arrays.asList(songCountPlayBean(songs.get(0), 6L),
+                              songCountPlayBean(songs.get(6), 6L),
+                              songCountPlayBean(songs.get(1), 5L),
+                              songCountPlayBean(songs.get(7), 5L),
+                              songCountPlayBean(songs.get(2), 4L)
+                              )
+                );
+
+  }
+
+  @Test
+  public void shouldDemonstrateInteractiveQueriesOnAnyValidHost() throws Exception {
```

Comment:
Similar to my feedback from the `WordCountInteractiveQueriesExampleTest`, do we need the full validation of IQ here, or is it sufficient to just verify the right host got bound?